Sources/Citadel/TTY/Client/TTY.swift

```diff
@@ -132,35 +142,30 @@ final class ExecCommandHandler: ChannelDuplexHandler {
     }
 
     func handlerRemoved(context: ChannelHandlerContext) {
-        onOutput(.eof(nil))
+        onOutput(context.channel, .eof(nil))
     }
 
     func channelRead(context: ChannelHandlerContext, data: NIOAny) {
         let data = self.unwrapInboundIn(data)
 
         guard case .byteBuffer(let buffer) = data.data else {
             logger.error("Unable to process channelData for executed command. Data was not a ByteBuffer")
-            return onOutput(.eof(SSHExecError.invalidData))
+            return onOutput(context.channel, .eof(SSHExecError.invalidData))
         }
         
         switch data.type {
         case .channel:
-            onOutput(.stdout(buffer))
+            onOutput(context.channel, .stdout(buffer))
         case .stdErr:
-            onOutput(.stderr(buffer))
+            onOutput(context.channel, .stderr(buffer))
         default:
             // We don't know this std channel
             ()
         }
     }
     
     func errorCaught(context: ChannelHandlerContext, error: Error) {
-        onOutput(.eof(error))
-    }
-
-    func write(context: ChannelHandlerContext, data: NIOAny, promise: EventLoopPromise<Void>?) {
-        let data = self.unwrapOutboundIn(data)
-        context.write(self.wrapOutboundOut(SSHChannelData(type: .channel, data: .byteBuffer(data))), promise: promise)
```

Comment:
@Joannis I've restructured and deduped the code significantly.
The Shell exec method now uses the same handler as the command exec and the pair method is also shared. In doing so this delegate method was causing some trouble as the channel request won't unwrap to ByteBuffer. It appears to be optional and if I understood it correctly it doesn't really do anything except for unwrapping and rewrapping the data. All methods still appear to work after removing this bit. WDYT?

Reply:
Update: I've reduced the code even further.